### ISSUE TITLE
Pushing a Character Map causes stack overflow

### ISSUE THREAD

**Issue by MattDahEpic** [loodakrawa/SpriterDotNet#76]
I'm attempting to change the player textures using a character map. The map works in Spriter but `UnityAnimator.SpriterProvider.PushCharMap()` throws a StackOverflowException. The error only occurs if the character map actually exists.
The bug occurs even in the example provided with SpriterDotNet.

My code:
```
animator.SpriteProvider.PushCharMap(GetMapByName(map_name,animator));
```
```
private static SpriterCharacterMap GetMapByName (string name, UnityAnimator animator) {
        foreach (SpriterCharacterMap map in animator.Entity.CharacterMaps) {
            if (map.Name.Equals(name)) return map;
        }
        GameRuntimeHelper.ThrowStoppingException("No character map exists for name "+name);
        return null;
    }
```
And the error:
```
StackOverflowException: The requested operation caused a stack overflow.
SpriterDotNet.Providers.DefaultAssetProvider`1[UnityEngine.Sprite].Get (Int32 folderId, Int32 fileId) (at Assets/SpriterDotNet/Lib/Providers/DefaultAssetProvider.cs:41)
SpriterDotNet.Providers.DefaultAssetProvider`1[UnityEngine.Sprite].Get (Int32 folderId, Int32 fileId) (at Assets/SpriterDotNet/Lib/Providers/DefaultAssetProvider.cs:41)
SpriterDotNet.Providers.DefaultAssetProvider`1[UnityEngine.Sprite].Get (Int32 folderId, Int32 fileId) (at Assets/SpriterDotNet/Lib/Providers/DefaultAssetProvider.cs:41)
...
and so on
```
which references [this line](https://github.com/loodakrawa/SpriterDotNet/blob/develop/SpriterDotNet/Providers/DefaultAssetProvider.cs#L41).

**Comment by loodakrawa** [comment]
I wasn't able to replicate the issue. The only case I can think of where this could be happening is when a character map file points to itself which shouldn't be happening. Is that perhaps the case?

Something like this:
`
<character_map id="0" name="alternate 1">
   <map folder="0" file="0" target_folder="0" target_file="0"/>
</character_map>
`

**Comment by MattDahEpic** [comment]
There are lines like that in the spriter file, is that a case of replacement textures being the same as the textures they replace?

**Comment by loodakrawa** [comment]
Depends on the content. It basically says - for this character map, replace [folder,file] with [target_folder,target_file]. So if they match it means the file points to itself which is causing the stack overflow. Can you please check if that's the case?

You also said it happens in the example - which modifications did you do to the example? Because I'm fairly certain it doesn't happen by default.

I'm trying to determine if this is just a case of weird data or an edge case I missed.

**Comment by MattDahEpic** [comment]
OK, I did a bit of testing. My animation was replacing sprites with themselves (folder==target_folder, file==target_file) and I replaced your GreyGuy with my Grey(AndGreen)Guy which had the same problem. 

The problem is my making, but if I could suggest some sort of sanity check when importing it may prevent future users from running into this problem since spriter seems to handle assigning textures to themselves just fine.
